src/grisp_connect_api.erl

```diff
@@ -1,152 +1,76 @@
 %% @doc Library module containing the jsonrpc API logic
 -module(grisp_connect_api).
 
--export([request/3]).
--export([notify/3]).
 -export([handle_msg/1]).
 
 -include_lib("kernel/include/logger.hrl").
 
 %--- Macros --------------------------------------------------------------------
--define(method_get, <<"get">>).
--define(method_post, <<"post">>).
--define(method_patch, <<"patch">>).
--define(method_delete, <<"delete">>).
+-define(method_get, get).
+-define(method_post, post).
 
-%--- API -----------------------------------------------------------------------
 
-% #doc Assembles a jsonrpc request and its uuid
--spec request(Method :: atom() | binary(),
-              Type :: atom() | binary(),
-              Params :: map()) -> {ID :: binary(), Encoded :: binary()}.
-request(Method, Type, Params) ->
-    ID = id(),
-    Rpc = {request, Method, maps:put(type, Type, Params), ID},
-    Encoded = grisp_connect_jsonrpc:encode(Rpc),
-    {ID, Encoded}.
+%--- API -----------------------------------------------------------------------
 
-% #doc Assembles a jsonrpc notification
--spec notify(Method :: atom() | binary(),
-              Type :: atom() | binary(),
-              Params :: map()) -> Encoded :: binary().
-notify(Method, Type, Params) ->
-    Rpc = {notification, Method, maps:put(type, Type, Params)},
-    grisp_connect_jsonrpc:encode(Rpc).
+% @doc Handles requests, notifications and errors from grisp.io.
```

Comment:
I see it only handling notifications and requests, not errors.

Reply:
It wasn't before the refactor IIRC, so I didn't add it.

Reply:
In fact, the request errors are handled in grisp_cllient: https://github.com/grisp/grisp_connect/blob/sylane/cleanup-connection/src/grisp_connect_client.erl#L260

Reply:
It was done before the refactoring (you can clearly see it in the diff) and it had a totally different documentation.
So that documentation is about your change and it doesn't fit. Fix it.

Reply:
You are right, and I remembered that and posted the second comment about where the error is handled. Now the error are handled in a generic way in grisp_connect_client, line 260